presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1247,29 +1238,41 @@ public int getResultSetHoldability()
     public int getDatabaseMajorVersion()
             throws SQLException
     {
-        // TODO: get version from server
-        return PrestoDriver.VERSION_MAJOR;
+        return getDatabaseVersionPart(0);
     }
 
     @Override
     public int getDatabaseMinorVersion()
             throws SQLException
     {
-        return PrestoDriver.VERSION_MINOR;
+        return getDatabaseVersionPart(1);
+    }
+
+    private int getDatabaseVersionPart(int part)
+            throws SQLException
+    {
+        String version = getDatabaseProductVersion();
+        List<String> parts = Splitter.on('.').limit(3).splitToList(version);
+        try {
+            return parseInt(parts.get(part));
+        }
+        catch (IndexOutOfBoundsException | NumberFormatException e) {
+            return 0;
```

Comment:
I just realized my comment for #1 is actually about the JDBC driver version, not the server version. However, the answer is nearly the same. The server version is determined the same way, from the package manifest, unless explicitly overridden with `presto.version`.